src/main/scala/onera/pmlanalyzer/pml/model/relations/AuthorizeRelation.scala

```diff
@@ -27,7 +27,7 @@ import onera.pmlanalyzer.pml.model.software.Application
   * @tparam L the left type
   * @tparam R the right type
   */
-case class AuthorizeRelation[L, R] private(iniValues: Map[L, Set[R]]) extends Relation[L, R](iniValues)
+final case class AuthorizeRelation[L, R] private(iniValues: Map[L, Set[R]]) extends Relation[L, R](iniValues)
```

Comment:
I will only tag one, but I suppose this is a semantic rule. If `AuthorizeRelation` was extended anywhere this would break. This raises two question:
1. In general, are such semantic rules ok knowing they will only have visibility of the pm-lanalyzer source code?
2. For the `final` case, there are many in this PR and are we sure non of those would need to be extended by users of the library?